infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/StreamsLogEntryReader.java

```diff
@@ -197,12 +204,17 @@ public LogReplicationEntry read(UUID logEntryRequestId) throws TrimmedException
                 }
 
                 lastOpaqueEntry = txOpaqueStream.next();
+                deltaCounter.ifPresent(Counter::increment);
+
                 lastOpaqueEntryValid = isValidTransactionEntry(lastOpaqueEntry);
                 currentProcessedEntryMetadata = new StreamIteratorMetadata(txOpaqueStream.txStream.pos(), lastOpaqueEntryValid);
             }
 
             log.trace("Generate LogEntryDataMessage size {} with {} entries for maxDataSizePerMsg {}. lastEntry size {}",
                     currentMsgSize, opaqueEntryList.size(), maxDataSizePerMsg, lastOpaqueEntry == null ? 0 : currentEntrySize);
```

Comment:
By looking at this, maybe another good metric to expose is in average how many opaque entries are sent per message... opaqueEntryList.size()

Reply:
Added the one that counts opaqueEntryList size.